File: Stacks/MinStack/min.py
class MinStack:

    def __init__(self):
        self.helper = []
        self.st = []

    def push(self, value: int) -> None:
        if len(self.st) == 0:
            self.st.append(value)
            self.helper.append(value)
            return
        val = self.helper[-1]
        if val > value:
            self.helper.append(value)
        else:
            self.helper.append(val)
        self.st.append(value)
        return

    def pop(self) -> None:
        if len(self.st)==0: return
        self.st.pop()
        self.helper.pop()
        return
        

    def top(self) -> int:
        return self.st[-1]

    def getMin(self) -> int:
        return self.helper[-1]
    
    

# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(value)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()

File: Stacks/MinStack/min.cpp
#include <stack>
using namespace std;

class MinStack1 {
public:
    stack<int>helper;
    stack<int>st;
    MinStack1() {
        
    }
    
    void push(int value) {
        if(st.empty()){
            st.push(value);
            helper.push(value);
            return;
        }
        int val = helper.top();
        if(val > value){
            helper.push(value);
        }
        else{
            helper.push(val);
        }
        st.push(value);
        return;
    }
    
    void pop() {
        if(st.empty())return;
        st.pop();
        helper.pop();
        return;
    }
    
    int top() {
        return st.top();
    }
    
    int getMin() {
        return helper.top();
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(value);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */